### ISSUE TITLE
Unexpected behavior when using `k` and `j` with count

### ISSUE THREAD

**Issue by thinety** [vim/vim#12675]
### Steps to reproduce

1. Create a file with many lines
2. Go to end of file and move cursor up with `50k` for example (50 is just a big number, greater than the window height)
3. Cursor jumps to middle of screen instead of staying on top
4. The same happens with `j` when moving down

### Expected behaviour

Expected `50k` to behave exactly the same as pressing `k` 50 times, but it does not: the latter stays at the top of the window (while also respecting configuration like `scrolloff`), while the former jumps to the middle of the screen. Analogously for `j`.

The behavior for `k` is caused by these lines:
https://github.com/vim/vim/blob/4c0089d696b8d1d5dc40568f25ea5738fa5bbffb/src/move.c#L395-L404
I can confirm that running the else block unconditionally solves the problem.

For `j` the logic is a bit more elaborate, I am only sure that it involves code around `scroll_cursor_halfway` in that same file.

Is this behavior expected or is it just an implementation detail?

### Version of Vim

9.0.1676

### Environment

OS: Arch Linux
Terminal: Alacritty
Shell: zsh

Default configuration (`vim --clean`)

**Comment by Pierre-Colin** [comment]
I get different behavior on both versions 9.0.1678 and 9.0.1683. Here are my steps:

- Run `vim --clean`.
- Run command `for i in range(1,200)|put=i|endfor` to generate a sequence of lines. The cursor is on line 200 (I’m ignoring the empty line before the enumeration starts).
- Press `100k`: the cursor is now on line 100, and the window is centered.
- Go to line 1 and press `100j`: the cursor is now on line 101 at the bottom of the window.

Not completely sure what the intended behavior is, but it does sound off that the window is centered when moving in one direction and not the other.

When you do it, the window is centered after both moves, right?

**Comment by chrisbra** [comment]
what is your terminal size please? For me with a default 80x24 terminal, in both directions the cursor is centered at line 100/101

**Comment by Pierre-Colin** [comment]
According to `tput`, 273×69.

**Comment by sukima** [comment]
Is it ok that it jumps to the middle? I found then when my cursor is at the bottom or top after a jump my first command is usually ZZ anyway.

**Comment by thinety** [comment]
> When you do it, the window is centered after both moves, right?

Yes, when using a 80x24 terminal. But I also get the behavior you described with a 273x69 terminal. Right now I am on version 9.0.1736.

> Is it ok that it jumps to the middle?

I would rather it behave the same as pressing `j` or `k` the same amount of times. But I wouldn't mind the cursor always jumping to the middle, as long as the behavior is consistent.

For some context, I ran into this issue because I am used to scrolling by moving the cursor. I have keymaps that move the cursor `'scroll'` lines up or down. The thing is that when "scrolling" up the cursor keeps jumping around, which is very distracting. When going down, it works fine.

![tmp34c7tovb-ascii](https://github.com/vim/vim/assets/51510921/48aa8575-b594-43c7-8733-87c82c4b34a9)

By experimenting, I found that increasing the number of lines (`:nmap J 20j`) makes the cursor also jump around when going down, and increasing it even more (`:nmap J 100j` and `:nmap K 100k`) keeps the cursor centered going both up and down. This was very surprising for me, ergo this issue.

What do you think should be the behavior in these situations?
